lib/rollbar/notifier.rb

```diff
@@ -675,34 +676,41 @@ def ssl_verify_mode
       end
     end
 
-    def write_item(item)
-      if configuration.use_async
-        @file_semaphore.synchronize do
-          do_write_item(item)
-        end
-      else
-        do_write_item(item)
-      end
-    end
-
     def do_write_item(item)
       log_info '[Rollbar] Writing item to file'
 
       body = item.dump
       return unless body
 
+      file_name = if configuration.files_with_pid_name_enabled
+                    configuration.filepath.gsub(EXTENSION_REGEXP, "_#{Process.pid}\\0")
```

Comment:
I'm not familiar with using the backslash zero in the filename here. Is this a convention?

Reply:
yeah, https://apidock.com/ruby/v2_6_3/String/gsub (see examples `Replacing with "\" and match`), it's just needed for replace `file_name.rollbar` to `file_name_12345.rollbar`
since in `filepath` includes extension and I have to do the replacement in this way(I specified that new flags only work if `.rollbar` is at the end, because it is hardcoded in a gem)

https://github.com/rollbar/rollbar-gem/blob/ce19357f2349d6d56426cc98341805ce8251a0dd/lib/rollbar/plugins/rails/railtie_mixin.rb#L17

Reply:
I see, it's a backreference for gsub. Thank you.